### PR TITLE
xtensa-build-zephyr.py: add -d option for debug build

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Add a new -d option to support SOF debug build. This
option will add CONFIG_DEBUG=y to SOF configuration,
which is aligned with xtensa-build-all.sh script.

Signed-off-by: Chao Song <chao.song@linux.intel.com>

### CODE REVIEW

File: scripts/xtensa-build-zephyr.py [hunk 1]
```diff
@@ -391,12 +393,14 @@ def build_platforms():
 		if args.cmake_args:
 			build_cmd += args.cmake_args
 
-		overlays = []
+		overlays = [str(item.resolve(True)) for item in args.overlay]
```
Comment (marc-hb) [inline scripts/xtensa-build-zephyr.py:398]: So much done in a single line and still very readable... impressive!

**Review by marc-hb — APPROVED** [review]
Approved assuming this has been tested with more than one `--overlay` option.

**Review by aborisovich — APPROVED** [review]

File: scripts/xtensa-build-zephyr.py (hunk 1 above)
Comment (aiChaoSONG) [reply scripts/xtensa-build-zephyr.py:398]: tested with `./scripts/xtensa-build-zephyr.py tgl -o overlays/tgl/ipc4_overlay.conf overlays/common/debug_overlay.conf`

**Review by marc-hb — CHANGES_REQUESTED** [review]
> tested with ./scripts/xtensa-build-zephyr.py tgl -o overlays/tgl/ipc4_overlay.conf overlays/common/debug_overlay.conf

Wait, I thought it would be `./scripts/xtensa-build-zephyr.py tgl -o overlays/tgl/ipc4_overlay.conf -o overlays/common/debug_overlay.conf` where each overlay is prefixed with `-o`. This avoids the ambiguity between positional and optional arguments. Also means additional overlays can be added by a script in any order with anything in between.

Can you please try `nargs=1, action="append"` instead?

**Pull request merged**